File: app/src/main/java/br/com/nicetuber/feature/home/HomeRouter.java
package br.com.nicetuber.feature.home;

import android.content.Context;
import android.content.Intent;

import org.parceler.Parcels;

import br.com.nicetuber.feature.channel_details.ChannelDetailsActivity;
import br.com.nicetuber.model.ChannelSearched;

import static br.com.nicetuber.util.ParamKey.KEY_CHANNEL;

/**
 * Created by thiagozg on 22/10/2017.
 */

public class HomeRouter {

    public static void goToChannelDetailsView(Context context, ChannelSearched channelSearched) {
        Intent intent = new Intent(context, ChannelDetailsActivity.class);
        intent.putExtra(KEY_CHANNEL, Parcels.wrap(channelSearched));
        context.startActivity(intent);
    }

}

File: app/src/main/java/br/com/nicetuber/feature/home/HomeRouter.kt
package br.com.nicetuber.feature.home

import android.app.Activity
import android.content.Intent
import br.com.nicetuber.feature.channel_details.ChannelDetailsActivity
import br.com.nicetuber.feature.channel_details.ChannelDetailsActivity.Companion.KEY_CHANNEL_SELECTED
import br.com.nicetuber.model.ChannelSearched

/**
 * Created by thiagozg on 22/10/2017.
 */

object HomeRouter {

    fun goToChannelDetailsView(activity: Activity, channelSearched: ChannelSearched) {
        val intent = Intent(activity, ChannelDetailsActivity::class.java)
        intent.putExtra(KEY_CHANNEL_SELECTED, channelSearched)
        activity.startActivity(intent)
    }

}
